tests/unit/test.config.js

```diff
@@ -0,0 +1,185 @@
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+          },
+        },
+      });
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, configObject.sourceDir);
+    });
+
+    it('preserves a string value on the command line over configured', () => {
+      const cmdLineSrcDir = '/user/specified/source/dir/';
+
+      const {argv, defaultValues} = makeArgv({
+        userCmd: ['fakecommand', '--source-dir', cmdLineSrcDir],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+          },
+        },
+      });
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, cmdLineSrcDir);
+    });
+
+    it('preserves configured value over default', () => {
+      const {argv, defaultValues} = makeArgv({
+        userCmd: ['fakecommand'],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, configObject.sourceDir);
+    });
+
+    it('preserves a string value on the command line over all others', () => {
+      const cmdLineSrcDir = '/user/specified/source/dir/';
+      const {argv, defaultValues} = makeArgv({
+        userCmd: ['fakecommand', '--sourceDir', cmdLineSrcDir],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const configObject = {
+        sourceDir: '/configured/source/dir',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, cmdLineSrcDir);
+    });
+
+    it('preserves default value of option if not in config', () => {
+      const {argv, defaultValues} = makeArgv({
+        userCmd: ['fakecommand'],
+        globalOpt: {
+          'source-dir': {
+            requiresArg: true,
+            type: 'string',
+            demand: false,
+            default: 'default/value/option/definition',
+          },
+        },
+      });
+      const configObject = {
+        foo: '/configured/foo',
+      };
+      const newArgv = applyConfigToArgv({argv, configObject, defaultValues});
+      assert.strictEqual(newArgv.sourceDir, argv.sourceDir);
```

Comment:
I would rather see `argv.sourceDir` replaced with `'default/value/option/definition'` to be sure it's getting the default.